fah_xchem/analysis/diffnet.py

```diff
@@ -262,12 +262,12 @@ def combine_free_energies(
             ]
         )
 
-        dgs = -gs - logsumexp(-gs)  # TODO: check math
+        dgs = gs - logsumexp(-gs)  # TODO: check math
         assert (dgs >= 0).all()
 
         for (node, _), dg in zip(valid_nodes, dgs):
             if node in supergraph:
-                supergraph.nodes[node]["g_exp"] = g_exp_compound + dg
+                supergraph.nodes[node]["g_exp"] = g_exp_compound - dg
```

Comment:
Sorry, I'm confused by this, but might be thinking about it backwards. I thought that the microstate free energies would be more positive than the compound free energy, representing occupancy that is a fraction of the compound? Isn't this what is described here? https://github.com/choderalab/fah-xchem/issues/80#issuecomment-692992085

Reply:
> This should be `+` if `dg > 0`

Do you mean "should be `-`" here? Your suggested diff changes to `-` from `+`.

Reply:
Sorry, I think I confused the K_d (dissociation constant) and K_a = 1/K_d (association constant) in the apportionment equation earlier. We need to split the K_a based on the normalized microstate probability. For example, if we have a K_a of 2*10^6 / M, and two equipotent microstates (p = 0.5), then each one contributes 1*10^6 / M to the K_a. In free energy, this would be
```
K_a = 2*10^6
g_exp = - ln (K_a / (1 M)) = - ln (2*10^6) = - 14
K_a_microstate = p_microstate * K_a = 0.5 * 2*10^6 / M = 1*10^6 / M
g_microstate = - ln (K_a_microstate / (1 M)) = -ln (1*10^6) = -13.8
```
So we need to 
* compute the normalized microstate `s` probability `p_is` for compound `i`
* convert that to a (positive) `dg_is`
* add that to the (negative) experimental `g_i` to get the `g_is > g_i`

Reply:
Thanks! I think this should be fixed in 41c0792.

Reply:
oops, pretty sure I had it wrong with the first attempt. The way in ef914d8 makes the most sense to me right now, and hopefully is also correct